sql/sqlserver/server.go

```diff
@@ -672,24 +674,38 @@ func (s *Server) processColumns(desc *structured.TableDescriptor,
 	return cols, nil
 }
 
-func (s *Server) processInsertRows(node parser.SelectStatement) (rows []sqlwire.Result_Row, err error) {
+func (s *Server) processSelect(node parser.SelectStatement) (rows []sqlwire.Result_Row, err error) {
 	switch nt := node.(type) {
+	// case *parser.Select:
+	// case *parser.Union:
+	// TODO(vivek): return s.query(nt.stmt, nil)
 	case parser.Values:
-		for _, row := range nt {
+		for _, tuple := range nt {
+			data, err := parser.EvalExpr(tuple, nil)
+			if err != nil {
+				return rows, err
+			}
+			dTuple, ok := data.(parser.DTuple)
+			if !ok {
+				// A one-element DTuple is currently turned into whatever its
+				// underlying element is, so we have to massage here.
+				// TODO(tschottdorf): Is that desired behaviour on behalf of
```

Comment:
Perhaps what we should be doing is passing to `EvalExpr` whether the caller wants a tuple or a single value.